skaffold.yaml

```diff
@@ -41,6 +41,11 @@ build:
         ldflags:
           - '{{.LD_FLAGS}}'
         main: ./cmd/gardener-extension-registry-cache
+      hooks:
+        after:
+          - command: 
+            - bash 
+            - hack/registry-cache-image-overwrite.sh
```

Comment:
I think with this change the `resourceSelector` section (L49-L53) can be removed. We no longer rely on the skaffold's feature to inject the image it built to the image field in the ControllerDeployment. We are generating the patch that updates the image values ourselves.